imports/plugins/core/hydra-oauth/server/oauthEndpoints.js

```diff
@@ -68,6 +68,19 @@ WebApp.connectHandlers.use("/consent", (req, res) => {
     .catch((errorMessage) => errorHandler(errorMessage, res));
 });
 
+WebApp.connectHandlers.use("/token/refresh", (req, res) => {
+  res.setHeader("Access-Control-Allow-Origin", process.env.OAUTH2_CLIENT_DOMAIN);
```

Comment:
According to this, you need the `Vary: Origin` header, too: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#CORS_and_caching

There would likely be many different domains hitting this. I believe the way this is usually done is setting the value to `req.headers.origin` after verifying that that origin is allowed.

Maybe change ENV name to DOMAINS plural, split it to an array on commas, and then check whether the request origin header is in that array?